internal/impl/kube.go

```diff
@@ -165,19 +171,23 @@ func GenerateKubeDeployment(image string, dep *protos.Deployment, cfg *KubeConfi
 	var generated []byte
 	generated = append(generated, content...)
 
-	// Generate the Jaeger deployment info.
-	content, err = generateJaegerDeployment(dep)
-	if err != nil {
-		return fmt.Errorf("unable to create kube jaeger deployment: %w", err)
+	if opts.Jaegar {
+		// Generate the Jaeger deployment info.
+		content, err = generateJaegerDeployment(dep)
+		if err != nil {
+			return fmt.Errorf("unable to create kube jaeger deployment: %w", err)
+		}
+		generated = append(generated, content...)
 	}
-	generated = append(generated, content...)
 
-	// Generate the Prometheus deployment info.
-	content, err = generatePrometheusDeployment(maps.Values(replicaSets), dep)
-	if err != nil {
-		return fmt.Errorf("unable to create kube deployment for the Prometheus service: %w", err)
+	if opts.Prometheus {
+		// Generate the Prometheus deployment info.
+		content, err = generatePrometheusDeployment(maps.Values(replicaSets), dep)
```

Comment:
This function also generates the ConfigMap for Prometheus. Please move the deployment to a separated function, and only turn it off.

Reply:
The reason for keeping ConfigMap is that it contains the targets to the services endpoint. And we can't be known after applying the change to update the config outside of the manifest.

Reply:
make sense. :D

Reply:
Ah, I see. Good point. I think if the user doesn't provide the `--prometheus` flag, then it is on them to set up Prometheus and make sure it works even when they change services. If the `--prometheus` flag is given, maybe there's something we can do to make that easier, but we can do that in a future PR.